---
title: "Rmd Workshop Assignment 1"
author: "Alissandra Hillis"
date: "`r Sys.Date()`"
output:
  html_document:
    toc: true
    toc_float: true
    code_folding: hide
---

# Set-up libraries and data

## Load libraries

```{r load-libraries}
library(tidyverse)
library(pheatmap)
```

## Load data

```{r load-data}
load("data/Rmarkdown_data.Rdata")
```

# Top 20 significant genes

## Get names of top 20 genes
	
```{r top20-names}
top20_sigOE_genes <- res_tableOE_tb %>% 
  arrange(padj) %>% 	#Arrange rows by padj values
  pull(gene) %>% 		#Extract character vector of ordered genes
  head(n=20)
```

## normalized counts for top 20 significant genes

```{r top20-counts}
top20_sigOE_norm <- normalized_counts %>%
  filter(gene %in% top20_sigOE_genes)
```

## Gathering the columns to have normalized counts to a single column

```{r gather-cols}
gathered_top20_sigOE <- top20_sigOE_norm %>%
  gather(colnames(top20_sigOE_norm)[2:9], key = "samplename", value = "normalized_counts")
gathered_top20_sigOE <- inner_join(mov10_meta, gathered_top20_sigOE)
```

## plot using ggplot2
In the plot below, you can see that most genes are upregulated with MOV10 overexpression, shown in blue, while MOV10 knockdown has little effect on gene expression for most genes.

```{r plot-top20}
ggplot(gathered_top20_sigOE) +
  geom_point(aes(x = gene, y = normalized_counts, color = sampletype)) +
  scale_y_log10() +
  xlab("Genes") +
  ylab("log10 Normalized Counts") +
  ggtitle("Top 20 Significant DE Genes") +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  theme(plot.title = element_text(hjust = 0.5))
```

# Create a heatmap of the differentially expressed genes

## Extract normalized expression for significant genes from the OE and control samples (2:4 and 7:9)

```{r get-normexp}
res_tableOE_tb_sig <- res_tableOE_tb %>%
  filter(padj < 0.05)
```

## Return the normalized counts for the significant DE genes

```{r get-DEgene-counts}
norm_OEsig <- normalized_counts %>% 
  filter(gene %in% res_tableOE_tb_sig$gene) 

meta <- mov10_meta %>%
column_to_rownames("samplename") %>%
data.frame()
```

## Run pheatmap using the metadata data frame for the annotation

In the heatmap below, you can again see that MOV10 overexpression results in significant upregulation of a subset of genes. As expected, there is little overlap in similarly regulated genes between MOV10 overexpression and knockdown.

```{r DEgene-heatmap}
pheatmap(norm_OEsig[2:9], 
         cluster_rows = T, 
         show_rownames = F,
         annotation = meta, 
         border_color = NA, 
         fontsize = 10, 
         scale = "row", 
         fontsize_row = 10, 
         height = 20)
```

## Session information
```{r session-info}
sessionInfo()
```
